### ISSUE TITLE
ARM64 Build

### ISSUE THREAD

**Issue by bodleytunes** [facebook/openr#11]
Does this build on ARM64?

**aarch64 / arm64** 

Latest and Ubuntu 16.04 on Scaleway

Compile error, seems to be a reference in folly build to some extensions that are not available.

g++: error: unrecognized command line option ‘-msse4.2’
g++: error: unrecognized command line option ‘-mpclmul’
Makefile:2178: recipe for target 'detail/libfollybasesse42_la-RangeSse42.lo' failed

Cheers,
Jon.

**Comment by saifhhasan** [comment]
We internally build for arm39x platform and works fine. However we do have some patches to make folly build happy.

Open task with folly https://github.com/facebook/folly for it ? I am sure you can figure it out.

**Comment by bodleytunes** [comment]
OK I will check it out :)

**Comment by saifhhasan** [comment]
Closing issue as it is not Open/R related.

**Comment by bodleytunes** [comment]
Just in case its of any use for someone in the future (n00bs to compiling like myself), I managed to compile on ARM64 on scaleway.

I had to download these files as it wouldn't detect the correct architecture:

wget http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD
wget http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD

config.sub and config.guess

I put these in /root/openr/build/deps/glog  replacing any old ones

I then had to remove references to -sse4.2 and the other flag in the CXX makefiles, e.g.  makefile.ac, makefile.in etc (just do a grep in the dir)

then add -Wno-noexcept-type into the CMAKe file as it was bombing out at some point during compilation.

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -Wno-noexcept-type")

Finally I was missing **libffi-dev**
apt install libffi-dev

Finally it compiled but with one error during the final tests.

It took hours to compile!

Cheers,
Jon.
